azure-proto-core-test/UserAssignedIdentityTests.cs

```diff
@@ -0,0 +1,377 @@
+﻿using azure_proto_core.Resources;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace azure_proto_core_test
+{
+    public class UserAssignedIdentityTests : UserAssignedIdentity
+    {
+        [TestCase(0, null, null, null, null)]
```

Comment:
Same comment as on SystemAssignedIdentities - make sure to exercise cases in which only one of the properties differs.

Reply:
Also, is it possible for client id or principal id to be null independently?